content/blog/architecture-as-code-vm/index.md

```diff
@@ -0,0 +1,286 @@
+---
+title: "Architecture as Code: Virtual Machines"
+date: 2020-03-31
+meta_desc: "Provisioning virtual machines and redeploying applications on existing instances using Pulumi"
+meta_image: vm.png
+authors:
+    - sophia-parafina
+tags:
+    - virtual machines
+    - provisioning
+---
+
+In a [previous article](https://www.pulumi.com/blog/architecture-as-code-intro/), we presented an overview of four infrastructure patterns for deploying modern applications. The article reviewed virtual machines, serverless, Kubernetes, and microservices. In this post, we'll examine virtual machines in-depth.
+
+<!--more-->
+
+## The Minimum Virtual Machine
+
+Whether you choose to build a virtual machine using VMWare or from a cloud service provider such as AWS, Azure, or Google Cloud Platform, they require three resources. First, the instance itself must be configured to meet application requirements. Second, network access for maintaining the instance and for applications. And finally, storage for application data.
+
+Let's start with the EC2 provisioner example, which is available on [Github](https://github.com/pulumi/examples/tree/master/aws-ts-ec2-provisioners). Make an empty directory and clone the example in the directory using `pulumi new`.
+
+```bash
+$ mkdir virtual_machine
```

Comment:
Sorry, missing this. Should we also include the `cd`?

```suggestion
$ mkdir virtual_machine && cd virtual_machine
```

Reply:
good catch